COLLABORATOR_GUIDE.md

```diff
@@ -391,6 +391,27 @@ Runtime Deprecations and End-of-life APIs (internal or public) must be
 handled as semver-major changes unless there is TSC consensus to land the
 deprecation as a semver-minor.
 
+Possible reasons for Runtime deprecations can be:
+
+* Node.js will no longer be able to support the API, for example if required
+  code is being removed in an upstream project.
+
+* The API is too hard or impossible to use correctly, increasing the risk of
+  undetected bugs or security issues in existing code.
+
+* Breaking the API makes way for simplification of its implementation that is
+  significant enough to make maintainability of Node.js easier, or enables
```

Comment:
> significant enough to make maintainability of Node.js easier

Compared to the ecosystem usage / taking ecosystem into account / something like that?

E.g. something huge and ugly but widely used is still kept, but something that adds only a bit of churn but isn't used by anyone can be runtime-deprecated quickly.

Reply:
@chalker I *think* this is at least implicitly covered by the paragraph below …

Either way, this doesn’t feel like the right place for that – this is listing the reasons to deprecate, and *all* decisions to deprecate should take ecosystem usage into account, right?